src/website/src/app/documentation/demos/datagrid/selection/selection.ts

```diff
@@ -56,13 +63,41 @@ export class DatagridSelectionDemo {
   rowSelectionExample = ROW_SELECTION_EXAMPLE;
   singleRowExample = SINGLE_ROW_EXAMPLE;
   selectionChangeEventExample = SELECTION_CHANGE_EVENT_EXAMPLE;
+  unselectableRow = UNSELECTABLE_ROW;
   users: User[];
   selected: User[] = [];
   rowSelected: User[] = [];
+  lockedUsers: User[] = [];
 
   constructor(inventory: Inventory) {
     inventory.size = 10;
     inventory.reset();
     this.users = inventory.all;
+
+    this.lockedUsers = [...inventory.all];
+  }
+
+  unlockRows() {
+    this.lockedUsers = this.lockedUsers.map(row => {
+      row.locked = undefined;
+      return row;
+    });
+  }
+
+  lockRows() {
+    // reset selections
+    this.selected = [];
+
+    this.lockedUsers = this.lockedUsers.map((user, index) => {
+      /** lock some rows */
+      if ([2, 3, 5, 9].includes(index)) {
+        user.locked = true;
+      }
+      /** also mark them as selected */
+      if ([2, 5].includes(index)) {
```

Comment:
Remove this selected state setting, the demo is confusing to lock and change selection at once.